forc-pkg/src/pkg.rs

```diff
@@ -235,6 +235,61 @@ impl BuildPlan {
         })
     }
 
+    /// Create a new build plan and syncronize manifest and lock file
```

Comment:
Might be useful to elaborate on this a little:

> Create a new build plan taking into account the state of both the `Manifest` and the existing lock file if there is one.
> 
> This will first attempt to load a build plan from the lock file and validate the resulting graph using the current state of the `Manifest`. This includes checking if the `[dependencies]` or `[patch]` tables have changed and checking the validity of the local `path` dependencies. If any changes are detected, the graph is updated and any new packages that require fetching are fetched.
> 
> The resulting build plan should always be in a valid state that is ready for `build`ing or `check`ing.

Reply:
Good catch, thanks this description is very clear. I'll add it in.